From 2e8121fb5bb84031ebedb669e8e096ff7903d842 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 31 Dec 2021 15:59:10 +0800
Subject: [PATCH] kabi: enables more stringent kabi checks
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 2e8121fb5bb84031ebedb669e8e096ff7903d842


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4K3S5

This option enables more stringent kabi checks. Those must be
disable in case of a debug-build because they allow to change struct
sizes.

We enable this option by default.

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/configs/openeuler_defconfig | 1 +
 arch/x86/configs/openeuler_defconfig   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index 63436eaa1ada..dd1bcf03e61a 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -7151,3 +7151,4 @@ CONFIG_SPFC=m
 CONFIG_GPIO_HISI=y
 CONFIG_OSNOISE_TRACER=y
 CONFIG_TIMERLAT_TRACER=y
+CONFIG_KABI_SIZE_ALIGN_CHECKS=y
diff --git a/arch/x86/configs/openeuler_defconfig b/arch/x86/configs/openeuler_defconfig
index e0dc1686984d..5d403124e49b 100644
--- a/arch/x86/configs/openeuler_defconfig
+++ b/arch/x86/configs/openeuler_defconfig
@@ -8529,3 +8529,4 @@ CONFIG_USERSWAP=y
 CONFIG_NET_VENDOR_RAMAXEL=y
 CONFIG_SPNIC=m
 CONFIG_SPFC=m
+CONFIG_KABI_SIZE_ALIGN_CHECKS=y
-- 
2.26.2

